corgi/core/models.py

```diff
@@ -592,7 +561,7 @@ class ProductVariant(ProductModel, TimeStampedModel):
 
     cpe = models.CharField(max_length=1000, default="")
 
-    product_variants = None  # type: ignore
+    product_variants = []  # type: ignore
```

Comment:
This behavior was copied from the original code, and isn't technically a bug, but I'm fine with changing it. Originally we had code in different models that was almost identical:
```
# No inheritance in original code
class ProductStream():
    products = ArrayField()
    product_versions = ArrayField()
    # product_streams field wasn't originally defined on ProductStream model
    product_variants = ArrayField()

# No inheritance in original code
class ProductVariant():
    products = ArrayField()
    product_versions = ArrayField()
    product_streams = ArrayField()
    # product_variants field wasn't originally defined on ProductVariant model
```

When I factored out that common code to the ProductModel class, I defined all four properties on the parent ProductModel class to avoid duplication and to make the schema generation work. To keep the behavior identical to the old code, I added e.g. ```product_variants = None``` so that the ProductVariant model still had the same properties as before.

In this case, setting ```product_variants = None``` means the field isn't defined on the model, even though it's defined in the parent class. So the field is "not present" and not "present but always None / not iterable". Maybe that's not the behavior we want, and we'd rather force all ProductModel subclasses to have the same properties.

In that case, ProductVariants here could just be removed (inherit the field definition from the parent class), or we could redefine the product_variants field on the ProductVariant model to make it read-only (force it to always be an empty ArrayField). Setting it to an empty list isn't quite what we want, I think. It's not "forcing the ArrayField to be initialized to an empty list" as I understand, but rather "replacing an ArrayField with a Python list" which won't define the Django methods for e.g. lookups.

Reply:
I'll switch it to ArrayField, and test removing the mytest skip flag as well. Thanks for explaining the history of the changes here.

Reply:
You're welcome - if you want it to be an ArrayField, you can just remove the line completely. It should get inherited from the parent ProductModel class.

Reply:
Switching to ArrayField would have required a migration, so I'm going to change it back to None, and just be careful about what we include in the ProductModel.builds properities so that we don't try to concatenate None to a list.

Reply:
To clarify a little more (probably overexplaining but want to be clear): it was set to None to avoid inheriting the property from the parent class. This was copied from the original code. We didn't define the "product_variants" field on the ProductVariant model, since it didn't make sense at the time.

In other words, this code defines what fields exist on the ProductVariant class. We aren't changing the value stored in the field or setting the "product_variants" field on any particular model instance. "None" here doesn't store a value inside the field, and you aren't trying to concatenate a None value to a list of values. "None" completely removes the field from the model definition. Then if you tried to add self.product_streams and self.product_variants from some ProductVariant instance, you would be trying to access a nonexistent product_variants property, which raises the below error:
```
In [1]: from corgi.core.models import *

In [2]: ProductVariant.objects.last()
Out[2]: <ProductVariant: Supplementary-9.0.0.Z.MAIN.EUS>

In [3]: ProductVariant.objects.last().product_streams
Out[3]: ['rhel-9.0.0.z']

In [4]: ProductVariant.objects.last().product_streams2
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [4], in <module>
----> 1 ProductVariant.objects.last().product_streams2

AttributeError: 'ProductVariant' object has no attribute 'product_streams2'
```

So removing "None" adds back the "product_variant" field the exact same way it's defined in the parent class. And changing what properties are defined on a model will always require a migration. We could make this change if we wanted to make the product_streams field on the ProductStream model default to a one-item list of [self.name], for example. Then all of our ProductModel subclasses would be consistent and have the same properties.